app/Http/Controllers/UserController.php

```diff
@@ -490,52 +491,39 @@ public function reset()
             }
         }
 
-        if (strtoupper($_SERVER['REQUEST_METHOD']) == 'POST' && isset($_POST['password']) && ! empty($_POST['password']) && isset($_POST['confirm_password']) && ! empty($_POST['confirm_password'])) {
-            $recovery = $_POST['recovery'];
-            $pwd = $_POST['password'];
-            $cpwd = $_POST['confirm_password'];
-            if (empty($recovery) || ! filter_var($recovery, FILTER_SANITIZE_STRING)) {
-                $response['danger'] = 'Recovery code invalid.';
+        $pwd = $request->post('password');
+        $cpwd = $request->post('confirm_password');
+        $response = null;
+        $email = null;
+
+        if ($request->getMethod() == 'POST' && $pwd && $cpwd) {
+            if (!$valid_code) {
+                $response['danger'] = __('passwords.token');
             } elseif ($pwd !== $cpwd) {
-                $response['danger'] = 'The passwords do not match';
+                $response['danger'] = __('passwords.match');
             } else {
-                $user = $User->where('recovery', '=', $recovery)->first();
-                if (! empty($user)) {
-                    $data = [
+                $email = $user->email;
+                $oldPassword = $user->password;
+
+                $update = $user->update([
                     'password' => crypt($pwd, '$1$'.strrev(md5(env('APP_KEY')))),
```

Comment:
Does this need to change to use the Laravel Hash helper?  I thought we changed it already recently - perhaps not.

Reply:
Perhaps yes.  Fix pushed.